### ISSUE TITLE
"Command failed with: 126"

### ISSUE THREAD

**Issue by lapsio** [CypherpunkArmory/UserLAnd#762]
So I tried to install UserLAnd in Work Profile and got following crash when filesystem was setting up:

![Screenshot_20190411-164218](https://user-images.githubusercontent.com/1054079/55967270-4f6fc000-5c7a-11e9-8a82-b6802fd9eb1c.png)

And then every attempt to run session ends up with:

![Screenshot_20190411-164659](https://user-images.githubusercontent.com/1054079/55967313-62829000-5c7a-11e9-9b99-48f6f0a35bad.png)

Where is mentioned error log placed?

# Describe the bug

Initial session setup crashes when app is installed in Work Profile.

# Steps to reproduce the behavior:

1. Set up Android Work Profile container.
2. Install UserLAnd in Work Profile container
3. Create new filesystem and session
4. Try to launch session

# Screenshots

![Screenshot_20190411-164218](https://user-images.githubusercontent.com/1054079/55967270-4f6fc000-5c7a-11e9-8a82-b6802fd9eb1c.png)

![Screenshot_20190411-164659](https://user-images.githubusercontent.com/1054079/55967313-62829000-5c7a-11e9-9b99-48f6f0a35bad.png)

# Device Information

Device: Asus ZS570KL.  
Android Version: 8.0.  
UserLAnd Version .  2.5.4
Whether the device is rooted.  no
Whether the device is running LineageOS. no

**Comment by lapsio** [comment]
added headers

**Comment by lapsio** [comment]
Itried to clean support files. After repeating procedure I didn't get "Permission denied" error. Only that "126" one.

**Comment by lapsio** [comment]
Okay I found log file, it contains 1 line:

    sh: support/execInProot.sh: Permission denied

**Comment by MatthewTighe** [comment]
Do things work normally while if the app is installed in a regular profile?

**Comment by lapsio** [comment]
> Do things work normally while if the app is installed in a regular profile?

I do have UserLAnd also installed in regular profile. Calling it "working normally" would be a bit of stretch but yes, it's _usually_ functional. At least session starts properly.

**Comment by lapsio** [comment]
I don't know how do you evoke scripts but maybe you have some hardcoded paths? Work Profile has different app data paths:

On top terminal in work profile, at the bottom terminal in normal profile. All paths typically containing "0" like `/storage/emulated/0` etc contain different numbers eg. 10

![Screenshot_20190411-184729](https://user-images.githubusercontent.com/1054079/55975464-530b4300-5c8a-11e9-92a7-92b920b7491a.png)

Maybe this information will be helpful.

**Comment by lapsio** [comment]
I don't know how do you evoke scripts but maybe you have some hardcoded paths? Work Profile has different app data paths:

On top terminal in work profile, at the bottom terminal in normal profile. All paths typically containing "0" like `/storage/emulated/0` etc contain different numbers eg. 10

![Screenshot_20190411-184729](https://user-images.githubusercontent.com/1054079/55975464-530b4300-5c8a-11e9-92a7-92b920b7491a.png)

Maybe this information will be helpful.

**Comment by lapsio** [comment]
I don't know how do you evoke scripts but maybe you have some hardcoded paths? Work Profile has different app data paths:

On top terminal in work profile, at the bottom terminal in normal profile. All paths typically containing "0" like `/storage/emulated/0` etc contain different numbers eg. 10

![Screenshot_20190411-184729](https://user-images.githubusercontent.com/1054079/55975464-530b4300-5c8a-11e9-92a7-92b920b7491a.png)

Maybe this information will be helpful.

**Comment by lapsio** [comment]
I don't know how do you evoke scripts but maybe you have some hardcoded paths? Work Profile has different app data paths:

On top terminal in work profile, at the bottom terminal in normal profile. All paths typically containing "0" like `/storage/emulated/0` etc contain different numbers eg. 10

![Screenshot_20190411-184729](https://user-images.githubusercontent.com/1054079/55975464-530b4300-5c8a-11e9-92a7-92b920b7491a.png)

Maybe this information will be helpful.

**Comment by lapsio** [comment]
I don't know how do you evoke scripts but maybe you have some hardcoded paths? Work Profile has different app data paths:

On top terminal in work profile, at the bottom terminal in normal profile. All paths typically containing "0" like `/storage/emulated/0` etc contain different numbers eg. 10

Maybe this information will be helpful.

**Comment by lapsio** [comment]
I don't know how do you evoke scripts but maybe you have some hardcoded paths? Work Profile has different app data paths:

On top terminal in work profile, at the bottom terminal in normal profile. All paths typically containing "0" like `/storage/emulated/0` etc contain different numbers eg. 10

![Screenshot_20190411-184729](https://user-images.githubusercontent.com/1054079/55975464-530b4300-5c8a-11e9-92a7-92b920b7491a.png)

Maybe this information will be helpful.

**Comment by MatthewTighe** [comment]
@corbinlc This looks like it could be the result from hardcoded paths in script shebangs. Any thoughts?

**Comment by corbinlc** [comment]
Yes, we don't work on anything but the main account/profile right now.  We have to patch the shebangs on a few scripts for it to work on other accounts.

**Comment by lapsio** [comment]
Thanks for quick diagnosis, is it fixable in any foreseeable future? I mean could you provide some info a about ETA or whether it's been added to TODO list for some release in near future?

**Comment by corbinlc** [comment]
Closing as part of a repository cleanup. If this issue is still relevant, feel free to reopen it or file a fresh one.
